core/src/commonMain/kotlin/com/powersync/db/PowerSyncDatabaseImpl.kt

```diff
@@ -87,7 +94,7 @@ internal class PowerSyncDatabaseImpl(
                 loggerFactory = loggerFactory
             )
 
-        scope.launch {
+        syncJob = scope.launch {
```

Comment:
We probably also need to call `disconnect` here before launching a new job. This is for cases where `connect` might be called multiple times. We want to ensure only 1 active connection is open at any point in time.